zproject/test_extra_settings.py

```diff
@@ -209,7 +209,7 @@ def set_loglevel(logger_name: str, level: str) -> None:
 VIDEO_ZOOM_CLIENT_ID = "client_id"
 VIDEO_ZOOM_CLIENT_SECRET = "client_secret"
 
-BIG_BLUE_BUTTON_SECRET = "123"
+BIG_BLUE_BUTTON_SECRET: Optional[str] = "123"
```

Comment:
mypy_django_plugin shouldn’t be looking at this file. I’m pretty sure we should be pointing it at `zproject.settings`, not `zproject.test_settings`.

Reply:
This was mainly because there are settings that are only available in `zproject.test_extra_settings` which the plugin don't have access to when starting from `zproject.settings`.

I guess we can add these to `zerver.computed_settings` and adjust the settings module?

```
[misc]  'Settings' object has no attribute 'APPLE_ID_TOKEN_GENERATION_KEY' 
[misc]  'Settings' object has no attribute 'BACKEND_DATABASE_TEMPLATE' 
[misc]  'Settings' object has no attribute 'CAMO_KEY' 
[misc]  'Settings' object has no attribute 'EXAMPLE_JWK' 
[misc]  'Settings' object has no attribute 'TEST_WORKER_DIR' 
```

Reply:
Actually, not all of them necessarily need to be in the settings. For example, Do we really need `EXAMPLE_JWK` as a setting?

Reply:
I see. `CAMO_KEY` is already in `computed_settings.py`, so I wonder why it’s complaining about that. The other four are test-only have no need to be in settings.

Reply:
`CAMO_KEY` depends on `CAMO_URI` to be set, otherwise it is not defined. We can as well just leave it `None` and make the setting `Optional`.

Reply:
Mypy doesn’t know about that dependency.

```python
CAMO_URI = ""
if CAMO_URI != "":
    CAMO_KEY = "secret"

reveal_type(CAMO_KEY) # note: Revealed type is "builtins.str"
```

It’s fine to make `CAMO_KEY` an `Optional` (in fact it already is, due to `get_secret`), but I’d like to understand what happened here.

Reply:
#22690

Reply:
> Mypy doesn’t know about that dependency.
> 
> ```python
> CAMO_URI = ""
> if CAMO_URI != "":
>     CAMO_KEY = "secret"
> 
> reveal_type(CAMO_KEY) # note: Revealed type is "builtins.str"
> ```
> 
> It’s fine to make `CAMO_KEY` an `Optional` (in fact it already is, due to `get_secret`), but I’d like to understand what happened here.

It looks like that mypy does not have the tightest type here:
```
reveal_type(CAMO_URI)  # Revealed type is "builtins.str"
```

Reply:
Because `mypy_django_stubs` does not initialize the settings with the overriden `CAMO_URI` and `CAMO_KEY` `zproject.test_extra_settings`, `CAMO_KEY` indeed does not exist on `django.conf.settings`. And because the plugin does this existence check solely based on the runtime information, but not the mypy symbol table which includes a symbol that is not eventually defined.